compiler/il/OMRNode.hpp

```diff
@@ -374,6 +374,23 @@ class OMR_EXTENSIBLE Node
       return true;
       }
 
+   static bool isNotDeprecatedUnsigned(TR::ILOpCodes opvalue)
+      {
+      
+      TR::ILOpCodes DeprecatedUnsignedList [] = 
+         {
+         TR::iumul, //Multiply
+         TR::buternary, TR::iuternary, TR::luternary, TR::suternary, //Ternary
+         TR::iushl, TR::lushl, //Shift
+         TR::ishfl, TR::lshfl, TR::iushfl, TR::lushfl, TR::bshfl, TR::sshfl, TR::bushfl, TR::sushfl //Shift Logical
+         };
+      int listSize = sizeof(DeprecatedUnsignedList)/sizeof(DeprecatedUnsignedList[0]);
+      for(int i = 0; i != listSize; i++)
+            if(opvalue == DeprecatedUnsignedList[i])
+               return false;
+      return true;
```

Comment:
Any reason why the list can't be static? There are also formatting problems (indentation, spaces after `for`, `if`). I also think this function can be simplified (have not tested the code):

```
static bool isNotDeprecatedUnsigned(TR::ILOpCodes opvalue)
   {
   static const TR::vector<TR::ILOpCodes> deprecatedUnsignedList =
      {
      TR::iumul, //Multiply
      TR::buternary, TR::iuternary, TR::luternary, TR::suternary, //Ternary
      TR::iushl, TR::lushl, // Shift
      TR::ishfl, TR::lshfl, TR::iushfl, TR::lushfl, TR::bshfl, TR::sshfl, TR::bushfl, TR::sushfl // Shift Logical
      };

   return std::find(deprecatedUnsignedList.begin(), deprecatedUnsignedList.end(), opvalue) != deprecatedUnsignedList.end();
   }
```

Reply:
@Leonardo2718 do you know if our compilers support something like this?

Reply:
Unfortunately, that kind of brace initialization is not supported by all our compilers [1].

An alternative that _should_ work is:

```c++
static const TR::ILOpCodes deprecatedUnsignedList[] =
   {
   TR::iumul, //Multiply
   TR::buternary, TR::iuternary, TR::luternary, TR::suternary, //Ternary
   TR::iushl, TR::lushl, // Shift
   TR::ishfl, TR::lshfl, TR::iushfl, TR::lushfl, TR::bshfl, TR::sshfl, TR::bushfl, TR::sushfl // Shift Logical
   };
      
   auto begin = &deprecatedUnsignedList[0];
   auto end = begin + sizeof(deprecatedUnsignedList)/sizeof(TR::ILOpCodes);
   return std::find(begin, end, opvalue) == end;
```

1. https://github.com/eclipse/omr/blob/master/doc/SupportedC%2B%2BFeatures.md#unsupported-c11-features

Reply:
This function is simplified in https://github.com/eclipse/omr/pull/4367/commits/5dfb0e229b4a249f4a26e1ee37632f69a237b9b4 as @Leonardo2718  has suggested.